b2sdk/_test_manager/api.py

```diff
@@ -0,0 +1,188 @@
+#
+# File: b2sdk/_test_manager/api.py
+#
+# Copyright 2023 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+import contextlib
+import random
+import string
+import time
+
+from datetime import datetime
+from os import environ
+from typing import Union
+
+import backoff
+
+from .bucket_tracking import BucketTrackingMixin
+from .._v3.exception import BucketIdNotFound as v3BucketIdNotFound
+from ..v2 import NO_RETENTION_FILE_SETTING, B2Api, Bucket, InMemoryAccountInfo, InMemoryCache, LegalHold, RetentionMode
+from ..v2.exception import BucketIdNotFound, DuplicateBucketName, FileNotPresent, TooManyRequests, NonExistentBucket
+
+ONE_HOUR_MILLIS = 60 * 60 * 1000
+ONE_DAY_MILLIS = ONE_HOUR_MILLIS * 24
+
+BUCKET_NAME_LENGTH = 50
+BUCKET_NAME_CHARS = string.ascii_letters + string.digits + '-'
+
+BUCKET_NAME_PREFIX = 'b2tst'
+
+# RUNNER_NAME is the only variable exposed by the GitHub CI that was changing for each matrix entry.
+# Example values are "GitHub Actions N" (with N being a whole number, starting from 2) and "Hosted Agent".
+# Here, we're using these names as long as time as seeds to start the random number generator.
+# Name fraction is used for runners inside the same matrix, time fraction is used for runners in different runs.
+# To avoid collision when the same runners are fired in different commits at the same time we also use GITHUB_SHA
+random.seed(
+    environ.get('RUNNER_NAME', 'local') + environ.get('GITHUB_SHA', 'local') + str(time.time_ns())
+)
+
+
+def generate_bucket_name() -> str:
+    suffix_length = BUCKET_NAME_LENGTH - len(BUCKET_NAME_PREFIX)
+    suffix = ''.join(random.choice(BUCKET_NAME_CHARS) for _ in range(suffix_length))
+    return f"{BUCKET_NAME_PREFIX}{suffix}"
+
+
+def current_time_millis() -> int:
+    return int(round(time.time() * 1000))
+
+
+class Api(BucketTrackingMixin, B2Api):
+    """
+    B2Api wrapper which should only be used for testing purposes!
+    """
+
+    def __init__(self, account_id: str, application_key: str, realm: str, *args, **kwargs):
+        info = InMemoryAccountInfo()
+        cache = InMemoryCache()
+        super().__init__(info, cache=cache, *args, **kwargs)
+        self.authorize_account(realm, account_id, application_key)
+
+    @backoff.on_exception(
+        backoff.constant,
+        DuplicateBucketName,
+        max_tries=8,
+    )
+    def create_test_bucket(self, bucket_type="allPublic", **kwargs) -> Bucket:
+        bucket_name = generate_bucket_name()
+        print('Creating bucket:', bucket_name)
+        try:
+            return self.create_bucket(bucket_name, bucket_type, **kwargs)
+        except DuplicateBucketName:
+            self._duplicated_bucket_name_debug_info(bucket_name)
+            raise
+
+    @backoff.on_exception(
+        backoff.expo,
+        TooManyRequests,
+        max_tries=8,
```

Comment:
Wouldn't it be better to extract file removal as a function/method and put a backoff on it? What do you think?